### DISCUSSION TITLE
.NET 8 MAUI - Codemagic Error Android SDK platform for API level 34 is not installed

### DISCUSSION THREAD

**Discussion by NiladriPadhy** [codemagic-ci-cd/codemagic-docs#2712] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?
Other - please mention in your question

.NET MAUI with .NET 8
### Steps to reproduce

Create a sample MAUI Multiplatform project with .NET 8.
I've followed the steps to build .NET MAUI apps as documented here: https://docs.codemagic.io/yaml-quick-start/building-a-dotnet-maui-app/ but with some changes as I'm using .NET 8 instead of .NET 6.

.NET 6 is outdated now. Please add a new migrattion document and sample .NET8 MAUI project.
### Expected results

Build should complete.

### Actual results

Build Error

[/Users/builder/clone/Readywire-MAUI.sln]
/Users/builder/clone/dotnet/packs/Microsoft.Android.Sdk.Darwin/34.0.130/tools/Xamarin.Android.Tooling.targets(100,5): error XA5207: Could not find android.jar for API level 34. This means the Android SDK platform for API level 34 is not installed; it was expected to be in `/usr/local/share/android-sdk/platforms/android-34/android.jar`. [/Users/builder/clone/ReadywireAttendance/ReadywireAttendance.csproj::TargetFramework=net8.0-android]
/Users/builder/clone/dotnet/packs/Microsoft.Android.Sdk.Darwin/34.0.130/tools/Xamarin.Android.Tooling.targets(100,5): error XA5207: You can install the missing API level by running `dotnet build -t:InstallAndroidDependencies -f net8.0-android "-p:AndroidSdkDirectory=/usr/local/share/android-sdk"`, or change the project to target an API version that is installed. [/Users/builder/clone/ReadywireAttendance/ReadywireAttendance.csproj::TargetFramework=net8.0-android]
/Users/builder/clone/dotnet/packs/Microsoft.Android.Sdk.Darwin/34.0.130/tools/Xamarin.Android.Tooling.targets(100,5): error XA5207: See https://aka.ms/xa5207 for more details. [/Users/builder/clone/ReadywireAttendance/ReadywireAttendance.csproj::TargetFramework=net8.0-android]

Workload updates are available. Run `dotnet workload list` for more information.

### Build id (optional)

66d62f7e7ecaff1628f2b701

**Comment by NiladriPadhy** [comment]
Similar issue closed as unanswered
https://github.com/orgs/codemagic-ci-cd/discussions/2255

**Reply by himesh-cm** [reply]
Hey @NiladriPadhy,

Please ssh into the build machine as shown [here](https://docs.codemagic.io/troubleshooting/accessing-builder-machine-via-ssh/) and verify that the `src` directory exists, then modify your command according to your project structure.

Please find the below error information
`line 9: cd: src: No such file or directory`

**Comment by NiladriPadhy** [comment]
Hi @himesh-cm  Seems like Android 34 is not available in Android SDK.

Get package "com.readywire.attendance" latest build number across all tracks
Unable to create an edit for package "com.readywire.attendance". The caller does not have permission
  Determining projects to restore...
  Restored /Users/builder/clone/ReadywireAttendance/ReadywireAttendance.csproj (in 32.46 sec).
/Users/builder/clone/dotnet/packs/Microsoft.Android.Sdk.Darwin/34.0.130/tools/Xamarin.Android.Tooling.targets(100,5): error XA5207: Could not find android.jar for API level 34. This means the Android SDK platform for API level 34 is not installed; it was expected to be in `/usr/local/share/android-sdk/platforms/android-34/android.jar`. [/Users/builder/clone/ReadywireAttendance/ReadywireAttendance.csproj::TargetFramework=net8.0-android]
/Users/builder/clone/dotnet/packs/Microsoft.Android.Sdk.Darwin/34.0.130/tools/Xamarin.Android.Tooling.targets(100,5): error XA5207: You can install the missing API level by running `dotnet build -t:InstallAndroidDependencies -f net8.0-android "-p:AndroidSdkDirectory=/usr/local/share/android-sdk"`, or change the project to target an API version that is installed. [/Users/builder/clone/ReadywireAttendance/ReadywireAttendance.csproj::TargetFramework=net8.0-android]
/Users/builder/clone/dotnet/packs/Microsoft.Android.Sdk.Darwin/34.0.130/tools/Xamarin.Android.Tooling.targets(100,5): error XA5207: See https://aka.ms/xa5207 for more details. [/Users/builder/clone/ReadywireAttendance/ReadywireAttendance.csproj::TargetFramework=net8.0-android]

Workload updates are available. Run `dotnet workload list` for more information.

**Reply by NiladriPadhy** [reply]
<img width="611" alt="Screenshot 2024-09-04 at 11 50 27 PM" src="https://github.com/user-attachments/assets/2d8a3a72-c5fa-4723-aa3f-42e04883e4d4">

I checked Android SDK folder of build machine, there android-34 folder is not exists. Attached screenshot for reference.

Even I added this script before build. Still getting above build error.
      - name: Android Build-tools
        script: |
          JAVA_HOME=/Library/Java/JavaVirtualMachines/zulu-8.jdk/Contents/Home
          $ANDROID_SDK_ROOT/tools/bin/sdkmanager --install "build-tools;34.0.0"

**Reply by NiladriPadhy** [reply]
@himesh-cm Could you please help me for this issue?

**Reply by NiladriPadhy** [reply, marked as answer]
Found the soluiton. Add below command before build starts. This will install Android 34 and other required tools.
 $DOTNET_BIN build -t:InstallAndroidDependencies -f net8.0-android "-p:AndroidSdkDirectory=/usr/local/share/android-sdk" -p:AcceptAndroidSdkLicenses=True
